modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/PreIssueAccessTokenTestCase.java

```diff
@@ -0,0 +1,470 @@
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_VIEW = "internal_org_user_mgt_view";
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_CREATE = "internal_org_user_mgt_create";
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_UPDATE = "internal_org_user_mgt_update";
+    private static final String INTERNAL_ORG_USER_MANAGEMENT_DELETE = "internal_org_user_mgt_delete";
+    private static final String INTERNAL_APPLICATION_MANAGEMENT_VIEW = "internal_application_mgt_view";
+    private static final String INTERNAL_APPLICATION_MANAGEMENT_UPDATE = "internal_application_mgt_update";
+    private static final String INTERNAL_API_RESOURCE_VIEW = "internal_api_resource_view";
+    private static final String INTERNAL_API_RESOURCE_CREATE = "internal_api_resource_create";
+    private static final String CUSTOM_SCOPE_1 = "test_custom_scope_1";
+    private static final String CUSTOM_SCOPE_2 = "test_custom_scope_2";
+    private static final String CUSTOM_SCOPE_3 = "test_custom_scope_3";
+
+    private static final String SCIM2_USERS_API = "/o/scim2/Users";
+    private static final String ACTIONS_API = "/api/server/v1/actions";
+    private static final String APPLICATION_MANAGEMENT_API = "/api/server/v1/applications";
+    private static final String API_RESOURCE_MANAGEMENT_API = "/api/server/v1/api-resources";
+
+    protected OAuth2ServiceHelper oAuth2Service;
+
+    protected SCIM2RestClient scim2RestClient;
+
+    private String accessToken;
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init(TestUserMode.TENANT_USER);
+
+        oAuth2Service = this.new OAuth2ServiceHelper();
+        scim2RestClient = new SCIM2RestClient(serverURL, tenantInfo);
+
+        setSystemproperties();
+
+        /* Pre Requisites */
+
+        ApplicationResponseModel application = oAuth2Service.addApplication(PASSWORD_GRANT_TYPE);
+
+        // creates systems APIs
+        if (!CarbonUtils.isLegacyAuthzRuntimeEnabled()) {
+            oAuth2Service.authorizeSystemAPIs(application.getId(), new ArrayList<>(Arrays.asList(SCIM2_USERS_API, ACTIONS_API,
+                    APPLICATION_MANAGEMENT_API, API_RESOURCE_MANAGEMENT_API)));
+        }
+
+        // creates domain APIs
+        List<String> customScopes = new ArrayList<>();
+        Collections.addAll(customScopes, CUSTOM_SCOPE_1, CUSTOM_SCOPE_2, CUSTOM_SCOPE_3);
+        String businessAPIId = oAuth2Service.createDomainAPIs(EXTERNAL_SERVICE_NAME, EXTERNAL_SERVICE_URI, customScopes);
+
+        // authorizes domain APIs
+        oAuth2Service.authorizeDomainAPIs(application.getId(), businessAPIId, customScopes);
+
+        // creates users and roles
+        List<Permission> permissions = addUserWithRole(application.getId(), customScopes);
+
+        // creates pre issue access token action type
+        createPreIssueAccessTokenAction();
+
+        // requests access token
```

Comment:
Since following `requestAccessToken` method uses `password` grant type, lets mention it here for better readbility
```suggestion
        // requests access token by password grant
```

Reply:
Updated with the commit 8bfab06a281f48e8f63c8f36b0555aae61a1e839